### PR TITLE
fix: zero the default Qwen3 classification bias

### LINKED ISSUE #38768: [Bug] Qwen3ForSequenceClassification retains random head bias when loading bias-free checkpoints
## Describe the bug

`Qwen3ForSequenceClassification` creates its classification head with `nn.Linear(config.hidden_size, config.num_labels)`, which enables and initializes a bias. Loading a checkpoint containing `score.weight` but no `score.bias` leaves that initialized bias unchanged and returns without an error.

The expected head computation for a bias-free checkpoint is `W @ h`; the resulting computation is `W @ h + b_init`. This can silently change classification logits.

The standard Transformers Qwen3 sequence-classification head uses `bias=False`, so a checkpoint without `score.bias` is a valid output of the reference implementation:

- [Transformers Qwen3 class](https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/models/qwen3/modeling_qwen3.py#L464-L465)
- [Inherited Transformers head constructor](https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/modeling_layers.py#L90-L99)
- [SGLang head constructor](https://github.com/sgl-project/sglang/blob/880d6fa64d187ae1d14b58101e2a776853492f26/python/sglang/srt/models/qwen3_classification.py)

## Reproduction

The CPU-only reproducer below executes the upstream head-construction statement, `Qwen3ForPooledOutput.load_weights`, and `default_weight_loader` unchanged, extracted from pinned public source with Python's AST module. It uses an eight-input, three-output synthetic head and a synthetic in-memory checkpoint.

This isolates head loading; it does **not** initialize the transformer backbone or start a SGLang server. Both the current-main revision `880d6fa64d187ae1d14b58101e2a776853492f26` and `v0.5.10.post1` reproduced the behavior.

Save the script below as `repro.py`, then run:

```sh
uv run --no-project --with torch==2.10.0 --with numpy python repro.py
uv run --no-project --with torch==2.10.0 --with numpy python repro.py v0.5.10.post1
```

<details>
<summary>Self-contained synthetic reproducer</summary>

```python
"""Exercise upstream Qwen3 head loading on CPU with synthetic tensors."""

import ast
import logging
import platform
import sys
import types
import urllib.request
from typing import Iterable, Tuple

import torch
from torch import nn

revision = sys.argv[1] if len(sys.argv) > 1 else '880d6fa64d187ae1d14b58101e2a776853492f26'
base = f'https://raw.githubusercontent.com/sgl-project/sglang/{revision}/python/sglang/srt/'


def source_tree(path):
    with urllib.request.urlopen(base + path, timeout=30) as response:
        return ast.parse(response.read().decode())


model_tree = source_tree('models/qwen3_classification.py')
classes = {node.name: node for node in model_tree.body if isinstance(node, ast.ClassDef)}
load_weights_node = next(
    node for node in classes['Qwen3ForPooledOutput'].body
    if isinstance(node, ast.FunctionDef) and node.name == 'load_weights'
)
constructor = next(
    node for node in classes['Qwen3ForSequenceClassification'].body
    if isinstance(node, ast.FunctionDef) and node.name == '__init__'
)
head_assignment = next(
    node for node in constructor.body
    if isinstance(node, ast.Assign) and ast.unparse(node.targets[0]) == 'self.score'
)
default_loader_node = next(
    node for node in source_tree('model_loader/weight_utils.py').body
    if isinstance(node, ast.FunctionDef) and node.name == 'default_weight_loader'
)

# Execute the upstream constructor statement and loader functions unchanged.
# The transformer backbone and server are intentionally outside this test.
namespace = dict(torch=torch, nn=nn, Iterable=Iterable, Tuple=Tuple,
                 logger=logging.getLogger('repro'))
exec(compile(ast.Module(body=[default_loader_node, load_weights_node],
                        type_ignores=[]), '<upstream>', 'exec'), namespace)
load_weights = namespace['load_weights']

torch.manual_seed(0)
model = nn.Module()
config = types.SimpleNamespace(hidden_size=8, num_labels=3)
exec(compile(ast.Module(body=[head_assignment], type_ignores=[]),
             '<upstream>', 'exec'), namespace | dict(self=model, config=config))
checkpoint = {'score.weight': torch.arange(24, dtype=torch.float32).reshape(3, 8) / 10}
initial_bias = model.score.bias.detach().clone()
load_weights(model, checkpoint.items())

inputs = torch.zeros(2, 8)
expected = torch.nn.functional.linear(inputs, checkpoint['score.weight'], bias=None)
actual = model.score(inputs)
assert torch.equal(model.score.weight, checkpoint['score.weight'])
assert torch.equal(model.score.bias, initial_bias)
assert torch.count_nonzero(model.score.bias) > 0
torch.testing.assert_close(actual - expected, initial_bias.expand_as(actual))

print(f'Python {platform.python_version()}, torch {torch.__version__}, CPU')
print(f'Upstream revision: {revision}')
print('Missing-bias load returned without error')
print('Retained initialized bias:', initial_bias.tolist())
print('Maximum logit error:', (actual - expected).abs().max().item())

load_weights(model, [('score.bias', torch.zeros(3))])
torch.testing.assert_close(model.score(inputs), expected, rtol=0, atol=0)
print('Explicit zero bias restores exact head-output parity')

supplied_bias = torch.tensor([0.1, -0.2, 0.3])
load_weights(model, [('score.bias', supplied_bias)])
torch.testing.assert_close(model.score(inputs), expected + supplied_bias)
print('Explicit nonzero checkpoint bias is loaded correctly')
```

</details>

Observed output for the current-main revision:

```text
Python 3.12.13, torch 2.10.0, CPU
Upstream revision: 880d6fa64d187ae1d14b58101e2a776853492f26
Missing-bias load returned without error
Retained initialized bias: [-0.05698692053556442, 0.037410933524370193, 0.3201341927051544]
Maximum logit error: 0.3201341927051544
Explicit zero bias restores exact head-output parity
Explicit nonzero checkpoint bias is loaded correctly
```

The release-tag run produced the same numerical results. The reported error is for this synthetic head, not a benchmark measurement.

## Expected behavior

- A standard bias-free checkpoint should retain its bias-free head computation.
- A supported checkpoint containing an explicit nonzero head bias should retain that bias.
- Add regression coverage for both cases so an absent tensor cannot leave an unintended initialized head bias behind.

One possible approach is an optional bias installed only when present in the checkpoint. The appropriate compatibility policy is up to the maintainers; the essential requirement is that this valid bias-free checkpoint not acquire an initialized bias.

## Environment

- Python: 3.12.13
- PyTorch: 2.10.0
- Execution: CPU on macOS arm64
- SGLang source tested: `v0.5.10.post1` and `880d6fa64d187ae1d14b58101e2a776853492f26` (current `main` when inspected)
- No full SGLang runtime is installed in this isolated reproducer, so there is no `sglang.check_env` output. This is a source-level/unit reproduction; GPU server behavior has not been exercised here.

I searched existing issues and PRs for `score.bias` and Qwen3 classification/bias reports and did not find this specific problem. #17992 addresses a different loading failure, and #30523 concerns repeated-request determinism rather than a retained bias after checkpoint loading.

### PR DESCRIPTION
**Pull request by git-jxj**
## Motivation

Fixes #38768. Standard Transformers Qwen3 sequence-classification checkpoints have `score.weight` but no `score.bias`. SGLang currently keeps the randomly initialized bias, so these checkpoints silently produce `W @ h + b_init` instead of `W @ h`.

## Modifications

Initialize the existing classifier bias to zero. Checkpoints that explicitly provide `score.bias` still overwrite it through the existing weight loader. Add a CPU CI regression exercising both checkpoint shapes through the real classification constructor and `load_weights`; the unit test isolates the unrelated backbone with `nn.Identity`.

## Accuracy Tests

- `PYTHONPATH=python python3 test/registered/unit/models/test_qwen3_classification_bias.py`: passed (missing-bias and explicit-bias subcases). The missing-bias case fails on upstream.
- Also instantiated the complete native one-layer Qwen3 model with a tiny config (`hidden_size=8`, `num_labels=3`) and a real single-rank Gloo process group, without mocking the backbone. Loaded `score.weight`, then compared classifier outputs on fixed hidden states with `torch.nn.functional.linear(..., bias=None)`. Maximum absolute logit error: **0.320134 before, 0.0 after**. The fixed model's classifier also passes on an RTX 5060 Ti (PyTorch 2.13.0+cu130).
- This validates head initialization/loading and head-output parity; it is not a full transformer-forward or model-server accuracy benchmark.
- Pre-commit checks passed for both changed files, including registered-test validation.

## Speed Tests and Profiling

The change only zeroes a small bias once during construction; the forward path is unchanged.

## Checklist

- [x] Format checks passed for changed files.
- [x] Added a registered CPU regression test using `CustomTestCase`.
- [x] Verified bias-free output parity and explicit checkpoint-bias compatibility.
- [x] Followed the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :x: [Run #34429353832](https://github.com/sgl-project/sglang/actions/runs/34429353832)
Latest PR Test (Extra): :x: [Run #34429353650](https://github.com/sgl-project/sglang/actions/runs/34429353650)
Latest PR Test (AMD ROCm 10): :x: [Run #34429353756](https://github.com/sgl-project/sglang/actions/runs/34429353756)

### CODE REVIEW

**Comment by git-jxj** [comment]
Current head `1d1a131df93d` passes lint, but the [normal PR CI gate](https://github.com/sgl-project/sglang/actions/runs/34429353832/job/102721419024) stops at `Missing required label 'run-ci'`, so the required `pr-test-finish` check has no successful test run. Could a maintainer or an authorized CI contributor authorize and rerun this head through the repository's `/tag-and-rerun-ci` flow? This account is not listed in `CI_PERMISSIONS.json`; an author-only rerun would still hit the same label gate.

**Comment by git-jxj** [comment]
Following up on my Sep 11 note: on current head `1d1a131df93d`, [`call-gate / pr-gate`](https://github.com/sgl-project/sglang/actions/runs/34429353832/job/102721419024) reports `Missing required label 'run-ci'`. The test jobs were skipped, so this head has no CI test result yet. Could an authorized CI user add the required label and trigger the tests for this head, and could a maintainer request a reviewer? Thanks.
